### PR TITLE
refactoring the job code

### PR DESCRIPTION
**Pull request by Sanjay1611**
This PR will create create a layer of `jobspec` between `job` and `podtemplatespec`

Signed-off-by: sanjay1611 <nathani.sanjay@mayadata.io>

### CODE REVIEW

**Review by ksatchit — DISMISSED** [review]

**Review by chandankumar4 — DISMISSED** [review]
Update the restart policy in chaos-scheduler and chaos-operator also check in chaos-exporter
/lgtm

**Comment by chandankumar4** [comment]
Add the comment for the exported function

**Review by rahulchheda — DISMISSED** [review]
lgtm!

File: kubernetes/jobspec/jobspec.go [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package jobspec
+
+import (
+	batchv1 "k8s.io/api/batch/v1"
+)
+
```
Comment (chandankumar4) [inline kubernetes/jobspec/jobspec.go:22]: Here also comment needed

**Review by chandankumar4 — APPROVED** [review]

**Pull request merged**